### PR TITLE
Feature implementation sst

### PR DESCRIPTION
**Pull request by inarteroger**
## Why is this change needed?

Adds the Spectral Space Transform (SST) transformer to the `chemotools.adaptation`
module. SST is a linear domain adaptation method that aligns target spectral data
to a source (reference) domain via SVD of the concatenated source-target matrix.
This extends the adaptation module alongside the existing DS and PDS transformers.

## Type of change

- [x] New feature

## Description

Implementation of `SpectralSpaceTransform` following the method described in:

> Du et al. (2011). Maintaining the predictive abilities of multivariate calibration models by spectral space transformation. 
> Analytica Chimica Acta, 690(1), 64–70.
> https://doi.org/10.1016/j.aca.2011.02.014

### What was added
- `SpectralSpaceTransform` transformer in `chemotools/adaptation/_spectral_space_transform.py`
- Full sklearn API compliance (`fit`, `transform`, `fit_transform`)
- Identity fallback when `X_source=None`

### Tests added
- sklearn estimator compliance
- fit attributes and shape checks
- boundary and edge case validation for `n_components`
- Pipeline + GridSearchCV + metadata routing integration

### CODE REVIEW

File: tests/adaptation/test_spectral_space_transform.py [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+        model2.fit(X_target, X_source=X_source)
+        result2 = model2.transform(X_test)
+
+        # Assert - Results should be bit-for-bit identical
+        np.testing.assert_array_equal(result1, result2)
+        np.testing.assert_array_equal(model1.p1_, model2.p1_)
+        np.testing.assert_array_equal(model1.p2_, model2.p2_)
+
+    def test_n_components_exceeds_n_samples(self):
+        """Verifies n_components > n_samples - 1 rises ValueError."""
```
[the remark this answers is not in the document]
Comment (inarteroger) [reply tests/adaptation/test_spectral_space_transform.py]: Done

File: chemotools/adaptation/_spectral_space_transform.py [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+    **Basic usage**
+    >>> import numpy as np
+    >>> from chemotools.adaptation import SpectralSpaceTransform
+    >>>
+    >>> rng = np.random.default_rng(17)
+    >>> X_source = rng.normal(size=(100, 20))
+    >>> X_target = X_source * 2 - rng.normal(size=(100, 20)) * 0.02
+    >>>
+    >>> sst = SpectralSpaceTransform(n_components=2).fit(X_target, X_source=X_source)
+    >>> X_transf = sst.transform(X_target)
```
[the remark this answers is not in the document]
Comment (inarteroger) [reply chemotools/adaptation/_spectral_space_transform.py:143]: Done

File: chemotools/adaptation/_spectral_space_transform.py [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+    ----------
+    n_components : int, default=2
+        Number of latent components retained from the singular value
+        decomposition (SVD). Controls the dimensionality of the shared subspace.
+
+
+    Attributes
+    ----------
+    n_features_in_ : int
+        Number of features seen during
```
[the remark this answers is not in the document]
Comment (inarteroger) [reply chemotools/adaptation/_spectral_space_transform.py]: Done

File: chemotools/adaptation/_spectral_space_transform.py [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+        assert self.p2_ is not None
+        # Compute the pseudo-inverse of p2_ to project from target latent space
+        # back to the feature space.
+        p2_inv = np.linalg.pinv(self.p2_)
+        # Apply the transformation:
+        # X @ p2_inv @ self.p1_ : map X to the latent space, then reconstruct
+        # in source domain
+        # X - (X @ p2_inv @ self.p2_) : add back the residual not captured by
+        # the latent space
+        return (X @ p2_inv @ self.p1_) + X - (X @ p2_inv @ self.p2_)
```
[the remark this answers is not in the document]
Comment (inarteroger) [reply chemotools/adaptation/_spectral_space_transform.py]: Done

File: chemotools/adaptation/_spectral_space_transform.py [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+        # Validate that n_components does not exceed the rank of X ( [X_source, X] )
+        # after centering.
+        # Centering reduces the effective rank in the sample direction by 1, so the
+        # maximum number of meaningful components is min(n_samples - 1, 2*n_features).
+        max_components = min(x.shape[0] - 1, x.shape[1])
+        if self.n_components > max_components:
+            raise ValueError(
+                f"n_components={self.n_components} is too large. "
+                f"After mean-centering, the effective rank of X is at most "
+                f"min(n_samples - 1, 2*n_features) = {max_components}. "
```
[the remark this answers is not in the document]
Comment (inarteroger) [reply chemotools/adaptation/_spectral_space_transform.py]: Done (modified the comments and the constraint/message)

File: chemotools/adaptation/_spectral_space_transform.py [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+            an identity transformation.
+
+        Returns
+        -------
+        self : SpectralSpaceTransform
+        """
+        # Validate the input parameters
+        # Check that X is a 2D array and has only finite values
+        X = validate_data(self, X, ensure_2d=True, reset=True, dtype=np.float64)
+
```
[the remark this answers is not in the document]
Comment (inarteroger) [reply chemotools/adaptation/_spectral_space_transform.py:195]: Done

File: chemotools/adaptation/_x_axis_interpolator.py [hunk 7]
```diff
@@ -99,11 +102,8 @@ def fit(self, X: np.ndarray, y=None, x_axis=None) -> "XAxisInterpolator":
         -------
         self : object
         """
```
[the remark this answers is not in the document]
Comment (paucablop) [reply chemotools/adaptation/_x_axis_interpolator.py:104]: No need because this is done in the decorator, it is a redundant operation

File: chemotools/adaptation/_spectral_space_transform.py [hunk 8]
```diff
@@ -0,0 +1,359 @@
+"""
+The :mod: `chemotools.adaptation._spectral_space_transform`
```
[the remark this answers is not in the document]
Comment (paucablop) [reply chemotools/adaptation/_spectral_space_transform.py]: Done

File: chemotools/adaptation/_piecewise_direct_standardization.py [hunk 9]
```diff
@@ -179,32 +192,61 @@ def fit(
...
-        self.coef_ = np.zeros((p, max_win), dtype=np.float64)
-        self.intercept_ = np.empty(p, dtype=np.float64)
+        # Pre-allocate a local build matrix (lil_matrix for sparse, ndarray for dense)
+        # and only assign to self.T_ after the final conversion, so the annotated
+        # type (ndarray | csr_matrix | None) is never violated mid-construction.
+        _T: np.ndarray | lil_matrix = (
+            lil_matrix((n_features, n_features), dtype=np.float64)
+            if self.storage == "sparse"
+            else np.zeros((n_features, n_features), dtype=np.float64)
+        )
```
[the remark this answers is not in the document]
Comment (paucablop) [reply chemotools/adaptation/_piecewise_direct_standardization.py:213]: Done
